tests/test_deploy_github_pages.py

```diff
@@ -282,8 +283,104 @@ def test_only_commit_dont_push(setup_test_env):
     remove_branch(target_branch)
 
 
-def test_selection_of_source_branch():
-    # at the moment requires you to be on source branch
-    # TODO implement once branch selection is implemented
-    pass
+def test_select_different_source_branch_which_does_exists_locally(setup_test_env):
```

Comment:
General questions, do we have tests for case, where it should fail

Reply:
No, but i can add some